tests/integration/test_sampling.py

```diff
@@ -368,3 +368,88 @@ def test_rate_limiter_on_long_running_spans(tracer):
 
     assert span_m29.context.sampling_priority > 0
     assert span_m30.context.sampling_priority > 0
+
+
+@pytest.mark.skipif(AGENT_VERSION != "testagent", reason="Tests only compatible with a testagent")
+@pytest.mark.snapshot()
+@pytest.mark.subprocess(
+    env={
+        "DD_SERVICE": "animals",
+        "_DD_TRACE_WRITER_NATIVE": "false",
```

Comment:
should we parametrize this as well to ensure it consistently gets applied with the native writer? in theory the behavior should be 1:1, no?

Reply:
The behavior with the native writer is bit complex: https://github.com/DataDog/dd-trace-py/blob/v3.12.1/ddtrace/_trace/processor/__init__.py#L164. If the native writer is used we avoid dropping spans in the SamplingProcessor even if stats computation is enabled. This changes the snapshot files and changes the sampling priority set on the chunk root span. I believe the writer should drop the spans but it doesn't (maybe this feature isn't complete yet).

I can add a test case that asserts the current behavior.